### ISSUE TITLE
chapter 3 Portable / Self-documenting

### ISSUE THREAD

**Issue by Pomar81** [MostlyAdequate/mostly-adequate-guide#551]
Is the next function is really pure?

```
const signUp = (Db, Email, attrs) => () => {
  const user = saveUser(Db, attrs);
  welcomeUser(Email, user);
};
```

**Comment by hugh-lx** [comment]
I think it's impure and we can use `memoize` to make a pure one.

**Comment by KtorZ** [comment]
Nope, this function is pure. Given the same inputs, it always return the same output. That output nevertheless, is a function (notice the `() =>...`).

That function with no arguments is what is sometimes called a "computation" or an "effect". Until the effect is executed (I.e. The function is run), it's completely pure. It's execution is not. There exists then abstractions on top of effects which allow transforming their results or combining them without executing them such that, a whole chain of computations can be constructed completely declaratively. 

For some languages like Haskell or Elm, the execution of effects is managed by the runtime. In Javascript, we resort to this trick and effects have to be manually executed
